src/COWShedFactory.sol

```diff
@@ -26,21 +33,27 @@ contract COWShedFactory is COWShedResolver {
         bytes calldata signature
     ) external {
         address proxy = proxyOf(user);
+        // deploy and initialize proxy if it doesnt exist
         if (proxy.code.length == 0) {
             COWShedProxy newProxy = new COWShedProxy{ salt: bytes32(uint256(uint160(user))) }(implementation, user);
             COWShed(payable(proxy)).initialize(address(this));
             emit COWShedBuilt(user, address(newProxy));
 
+            // set reverse mapping of proxy to owner
             ownerOf[proxy] = user;
 
+            // if on mainnet, set the forward and reverse resolution nodes
             if (block.chainid == 1) {
                 _setReverseNode(user, proxy);
                 _setForwardNode(user, proxy);
             }
         }
+        // execute the hooks
```

Comment:
Would recommend a note here that this can't be trusted execute as this is critical to avoid arbitrary execution on a proxy from someone who shouldn't be the owner (i.e. authorisation happens in the following call to authorise the calls.